bin/mocha

```diff
@@ -151,5 +152,5 @@ if (Object.keys(nodeArgs).length) {
     proc.kill('SIGTERM'); // if that didn't work, we're probably in an infinite loop, so make it die.
   });
 } else {
-  require('../lib/cli/cli').main(unparse(mochaArgs, {alias: aliases}));
+  esmRequire('../lib/cli/cli').main(unparse(mochaArgs, {alias: aliases}));
```

Comment:
In my tests this didn't work. Overriding `require` with `require = require('esm')(module);` on L13 does work though.
Maybe there are more solutions.
Just loading the module `esm` as we do it in [lib/cli/run-helpers.js](https://github.com/mochajs/mocha/blob/master/lib/cli/run-helpers.js#L82) is not working, additionally the `require` function has to be overridden, somewhere ...

Reply:
> In my tests this didn't work.

How were you testing? Seemed ok for me (for core code).

Reply:
I just tested this test file:
```js
import {ok} from 'assert';
describe('fixture', function() {
  it('should be true', function() {
    ok(true);
  });
});
```
It does not run with the changes of this PR. It does run with `--require esm`.

Reply:
Yeah I think `esmRequire ` only gets applied to core code.